### DISCUSSION TITLE
How to check user group for multiple methods at once?

### DISCUSSION THREAD

**Discussion by cornejobarraza** [codeigniter4/shield#598] [Q&A] [answered]
Hi everyone!

Shield has helped me so much in a project I'm working on right now but I'm not sure about the following 🤔

Right now I'm passing a check to all my methods inside my controller, for example:

```php
$user = auth()->user();

if ($user->inGroup('admin')) {
     return view('management/home');
} else {
     return redirect()->to(base_url('/'))->with('error', 'You have no permission to see this page...');
}
```

This works great but I have to repeat myself for every method I have...

Is there a way I can perform this check for multiple methods inside a controller? Thanks!

**Comment by datamweb** [comment, marked as answer]
Hi, Thank you for your positive energy.

The method you suggested works, but it is **not the most correct method.**

We have [6 filters](https://github.com/codeigniter4/shield/blob/develop/docs/install.md#controller-filters) in Shield, in this regard you should use filter **group**.

Everything you need is explained [Authorizing via Routes](https://github.com/codeigniter4/shield/blob/develop/docs/authorization.md#authorizing-via-filters)

> **Note** There are different ways to use filters in **CodeIgniter4**, you can read the full description of how to use filters [filters](https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html?highlight=filter).

 if you think it is still unclear let us know.

**Reply by cornejobarraza** [reply]
Yes that worked! I didn't notice I was getting Shield from the beta releases instead of dev-develop, so it wasn't working because some files were missing.

Also, in the filters file I had to do it like this so it worked, otherwise it said an alias was missing for the filter:

```
public $filters = [
    // ...
    'group' => ['before' => ['admin/*']],
];
```

And then on the routes I specified `group:admin,superadmin` for the filter as specified on the docs:

```
// ...
$routes->group('admin', ['filter' => 'group:admin,superadmin'], static function ($routes) {
    $routes->get('users', 'Management\Users::index');
});
```

**Reply by datamweb** [reply]
> Also, in the filters file I had to do it like this so it worked, otherwise it said an alias was missing for the filter:

Documents were updated in this regard. Thanks. see #600

**Reply by cornejobarraza** [reply]
> > Also, in the filters file I had to do it like this so it worked, otherwise it said an alias was missing for the filter:
> 
> Documents were updated in this regard. Thanks. see #600 

Glad to be helping a bit, thanks for all the great resources and Shield!
